### ISSUE TITLE
DirectoryNotEmptyException

### ISSUE THREAD

**Issue by pihme** [zeebe-io/zeebe#5513]
**Describe the bug**
Observed as error in production logs.

**Log/Stacktrace**
https://console.cloud.google.com/errors/CJWc79PagJbCXQ?service=zeebe&time=P7D&project=camunda-cloud-240911

**Environment:**
Camunda Cloud / 0.24.2, 0.24.3

**Comment by deepthidevaki** [comment]
Version: zeebe:0.25.0 
```
 java.nio.file.DirectoryNotEmptyException: /usr/local/zeebe/data/raft-partition/partitions/2/pending/12499-56-1604939548411-13039-13040.tmp

        at sun.nio.fs.UnixFileSystemProvider.implDelete (Unknown Source)
        at sun.nio.fs.AbstractFileSystemProvider.delete (Unknown Source)
        at java.nio.file.Files.delete (Unknown Source)
        at io.zeebe.util.FileUtil$1.postVisitDirectory (FileUtil.java:67)
        at io.zeebe.util.FileUtil$1.postVisitDirectory (FileUtil.java:56)
        at java.nio.file.Files.walkFileTree (Unknown Source)
        at java.nio.file.Files.walkFileTree (Unknown Source)
        at io.zeebe.util.FileUtil.deleteFolder (FileUtil.java:54)
        at io.zeebe.snapshots.broker.impl.FileBasedSnapshotStore.purgePendingSnapshot (FileBasedSnapshotStore.java:314)

```

**Comment by npepinpe** [comment]
We observed this occurs during transitions, so the assumption is the impact is quite low (as the node could not commit this snapshot anyway as it's transitioning) and the pending snapshot is later removed normally.

**Comment by deepthidevaki** [comment]
This should be fixed by #6283
Please reopen if occured again.
